### DISCUSSION TITLE
Damageable drawing performance

### DISCUSSION THREAD

**Discussion by SomeRandomNoobKekeke** [FakeFishGames/Barotrauma#15103] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

If you're wondering why ruins are lagging in lighting and damageable on high zoom, then it's because of this insertion sort

https://github.com/FakeFishGames/Barotrauma/blob/7d5b7a310a52372dac0a8bac90fd2a6a2ab342dc/Barotrauma/BarotraumaClient/ClientSource/Map/Submarine.cs#L171-L190
![20241110232656_1](https://github.com/user-attachments/assets/4d178b25-fcca-42c1-b28e-3ad23c82d96f)

it's substantial even on normal zoom
![20241110235608_1](https://github.com/user-attachments/assets/77fdce2f-97d2-4fdf-bc10-39b84f3d6f07)
It doesn't really depend on predicate, it uses it only once in sub editor:

https://github.com/FakeFishGames/Barotrauma/blob/7d5b7a310a52372dac0a8bac90fd2a6a2ab342dc/Barotrauma/BarotraumaClient/ClientSource/Screens/SubEditorScreen.cs#L6250
### So
Are you sure you need to sort them every draw frame? 
Are you sure you want to sort them with insertions?
Are you sure you need to do same work twice in DrawMap and in RenderLightMap?

### Reproduction steps

check ruins, or anything else

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Single player

### Version

v1.6.19.1 (Unto the Breach Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
This has been addressed in the rendering optimizations included in the latest unstable build.
